### PR TITLE
database extension checking improvements

### PR DESCRIPTION
**Pull request by acesnik**

### CODE REVIEW

File: GUI/XLSearchTaskWindow.xaml [hunk 1]
```diff
@@ -63,36 +63,33 @@
                                     <TextBox x:Name="txtXLBetaPrecusorMsTl" Width="45" IsEnabled="{Binding IsChecked, ElementName=ckbSearchWithXLAllBeta}" />
                                     <ComboBox x:Name="cbbXLBetaprecusorMsTl" IsEnabled="{Binding IsChecked, ElementName=ckbSearchWithXLAllBeta}" />
                                 </StackPanel>-->
-                                <StackPanel Orientation="Horizontal" Grid.Row="1" Grid.Column="0">
-                                    <Label x:Name="lblXLprecusorMsTl"  Content="Crosslink Precusor mass tolerance" />
+                                <StackPanel Orientation="Horizontal" Grid.Row="1" Grid.Column="0" Margin="5">
```
Comment (rmillikin) [inline GUI/XLSearchTaskWindow.xaml]: I think you're not synced up or something, these are changes I made to the XL Search Task window recently

File: GUI/MainWindow.xaml.cs [hunk 2]
```diff
@@ -383,7 +383,9 @@ private void AddAFile(string draggedFilePath)
             // we need to get the filename before parsing out the extension because if we assume that everything after the dot
             // is the extension and there are dots in the file path (i.e. in a folder name), this will mess up
             var filename = Path.GetFileName(draggedFilePath);
-            var theExtension = filename.Substring(filename.IndexOf(".")).ToLowerInvariant();
+            var theExtension = Path.GetExtension(filename).ToLowerInvariant();
```
Comment (rmillikin) [inline GUI/MainWindow.xaml.cs]: this looks fine, you will also need to change code in MetaMorpheus/TaskLayer/GPTMDTask/GPTMDTask.cs line ~201 for GPTMD database writing to make sure the correct file extension (.xml) is written there and not something weird like .xml.xml when searching with an .xml.gz

File: TaskLayer/MetaMorpheusTask.cs [hunk 3]
```diff
@@ -301,7 +301,11 @@ protected static List<Protein> LoadProteinDb(string fileName, bool generateTarge
             List<string> dbErrors = new List<string>();
             List<Protein> proteinList = new List<Protein>();
 
-            if (Path.GetExtension(fileName).Equals(".fasta") || Path.GetExtension(fileName).Equals(".fa"))
+            var theExtension = Path.GetExtension(fileName).ToLowerInvariant();
+            bool compressed = theExtension.EndsWith("gz"); // allows for .bgz and .tgz, too which are used on occasion
```
Comment (rmillikin) [inline TaskLayer/MetaMorpheusTask.cs]: can you do a search to check that MetaMorpheus can handle .fa.gz and .fasta.gz? and preferably add that to a unit test

**Comment by trishorts** [comment]
would be great if it could read .raw.gz and .mzml.gz

**Comment by rmillikin** [comment]
Anthony mentioned that compressed spectra files may have artefacts due to compression. I'm not sure what the details are.

Can I close this? seems like all? of these changes were made in the other PR

**Comment by acesnik** [comment]
Yes

**Pull request closed without merging**